File: wow-minmax-ui/src/app/modules/character/model/equipment/GemOptions.ts
import { EquippableItem } from "./EquippableItem";
import { Gem } from "./Gem";
import { SocketType } from "./SocketType";

export interface GemOptions {
	socketType: SocketType;
	gems: Gem[];
}

export function getMatchingGemOptions(gemOptions: GemOptions[], equippedItem: EquippableItem, socketIdx: number) {
	const socketType = equippedItem.item.socketTypes[socketIdx];
	return gemOptions.find(x => x.socketType === socketType)?.gems || []
}

File: services/wow-minmax/src/main/java/wow/minmax/model/GemOptions.java
package wow.minmax.model;

import wow.commons.model.item.Gem;
import wow.commons.model.item.SocketType;

import java.util.List;

/**
 * User: POlszewski
 * Date: 2025-08-30
 */
public record GemOptions(
		SocketType socketType,
		List<Gem> gems
) {
}
